File: src/test/java/org/projectodd/nodej/PathTest.java
package org.projectodd.nodej;

import static org.fest.assertions.Assertions.assertThat;

import java.io.IOException;

import org.dynjs.runtime.DynObject;
import org.dynjs.runtime.Types;
import org.junit.Before;
import org.junit.Test;

public class PathTest extends NodejTestSupport {
    
    @Before
    public void beforeTests() {
        assertThat(eval("path = require('path'); path").getClass()).isEqualTo(DynObject.class);
    }

    @Test
    public void testNormalize() {
        assertThat(eval("path.normalize('/foo/bar//baz/asdf/quux/..')")).isEqualTo("/foo/bar/baz/asdf");
    }
    
    @Test
    public void testJoin() {
        assertThat(eval("path.join('/foo', 'bar', 'baz/asdf', 'quux', '..')")).isEqualTo("/foo/bar/baz/asdf");
        assertThat(eval("path.join('foo', {}, 'bar')")).isEqualTo("foo/bar");
    }
    
    @Test
    public void testResolve() {
        assertThat(eval("path.resolve('/foo/bar', './baz')")).isEqualTo("/foo/bar/baz");
    }
    
    @Test
    public void testRelative() {
        assertThat(eval("path.relative('/data/orandea/test/aaa', '/data/orandea/impl/bbb')")).isEqualTo("../../impl/bbb");
    }
    
    @Test
    public void testDirname() {
        assertThat(eval("path.dirname('/foo/bar/baz/asdf/quux')")).isEqualTo("/foo/bar/baz/asdf");
    }
    
    @Test
    public void testBasename() {
        assertThat(eval("path.basename('/foo/bar/baz/asdf/quux.html')")).isEqualTo("quux.html");
        // TODO: Not clear why this is failing
//        assertThat(eval("path.basename('/foo/bar/baz/asdf/quux.html', '.html')")).isEqualTo("quux");
    }
    
    @Test
    public void testExtname() {
        assertThat(eval("path.extname('index.html')")).isEqualTo(".html");
    }
    
    @Test
    public void testPathSep() {
        assertThat(eval("path.sep")).isEqualTo(System.getProperty("file.separator"));
    }
}

File: src/test/resources/path/path_test.js
var vertxTest = require('vertx_tests');
var vassert   = vertxTest.vassert;

var path = require('path');

var PathTests = {
  testNormalize: function() {
    vassert.assertEquals("/foo/bar/baz/asdf", path.normalize('/foo/bar//baz/asdf/quux/..'));
    vassert.testComplete();
  },

  testJoin: function() {
    vassert.assertEquals("/foo/bar/baz/asdf", path.join('/foo', 'bar', 'baz/asdf', 'quux', '..'));
    vassert.assertEquals("foo/bar", path.join('foo', {}, 'bar'));
    vassert.testComplete();
  },

  testResolve: function() {
    vassert.assertEquals("/foo/bar/baz", path.resolve('/foo/bar', './baz'));
    vassert.testComplete();
  },

  testRelative: function() {
    vassert.assertEquals("../../impl/bbb", path.relative('/data/orandea/test/aaa', '/data/orandea/impl/bbb'));
    vassert.testComplete();
  },

  testDirname: function() {
    vassert.assertEquals("/foo/bar/baz/asdf", path.dirname('/foo/bar/baz/asdf/quux'));
    vassert.testComplete();
  },

  testBasename: function() {
    vassert.assertEquals("quux.html", path.basename('/foo/bar/baz/asdf/quux.html'));
    vassert.assertEquals("quux", path.basename('/foo/bar/baz/asdf/quux.html', ".html"));
    vassert.testComplete();
  },

  testExtname: function() {
    vassert.assertEquals(".html", path.extname('index.html'));
    vassert.testComplete();
  },

  testPathSep: function() {
    vassert.assertEquals(java.lang.System.getProperty("file.separator"), path.sep);
    vassert.testComplete();
  }
}

vertxTest.startTests(PathTests);
